envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadata.kt

```diff
@@ -54,6 +74,33 @@ private fun getCommunicationMode(proto: Value?): CommunicationMode {
     }
 }
 
+fun Value?.toStatusCodeFilter(properties: AccessLogFilterProperties): AccessLogFilter.StatusCodeFilter? {
+    val value = this?.stringValue
+
+    if (value != null) {
+        var op: ComparisonFilter.Op? = null
+        val regex = """((le)|(eq)|(ge)){1}:(\d{3})""".toRegex()
+
+        val matchResult = regex.matchEntire(value.toLowerCase())
+        matchResult?.takeIf { !it.groupValues.isEmpty() }?.apply {
+            when (matchResult.groupValues.get(properties.operatorIndex)) {
+                "le" -> op = ComparisonFilter.Op.LE
+                "eq" -> op = ComparisonFilter.Op.EQ
+                "ge" -> op = ComparisonFilter.Op.GE
+            }
+            return AccessLogFilter.StatusCodeFilter(
+                comparisonOP = op!!,
+                comparisonCode = matchResult.groupValues.get(properties.codeIndex).toInt()
+            )
+        } ?: run {
+            throw NodeMetadataValidationException(
+                "Access log filter status code doe not match pattern: ((le)|(eq)|(ge)){1}:(\\d{3})"
+            )
+        }
```

Comment:
You don't need `run` here.
```suggestion
        } ?: throw NodeMetadataValidationException(
            "Access log filter status code doe not match pattern: ((le)|(eq)|(ge)){1}:(\\d{3})"
        )
```